services/server/server.go

```diff
@@ -255,11 +258,27 @@ func (s *Server) Version(ctx context.Context, req *serverpb.VersionRequest) (*se
 // Readiness returns an error when some PMM Server component is not ready yet or is being restarted.
 // It can be used as for Docker health check or Kubernetes readiness probe.
 func (s *Server) Readiness(ctx context.Context, req *serverpb.ReadinessRequest) (*serverpb.ReadinessResponse, error) {
-	// TODO https://jira.percona.com/browse/PMM-1962
 
-	if err := s.prometheus.Check(ctx); err != nil {
+	errs := make([]error, 0)
```

Comment:
It looks too complicated.
We don't have to return all errors. We can return only first error.

Reply:
I think we should return all errors at once to make debugging easier and faster otherwise, supose both, prometheus and grafana are failing, you would need to run the check twice to get both errors.

Reply:
We usually need it for pmm-update and healthcheck, so user don't fix it manually. Am I right, @AlekSi

Reply:
In general, I agree with @percona-csalguero.

But in that particular case we should expose less information, not more – that API is available without authentication:
```proto
message ReadinessRequest {}

message ReadinessResponse {
  // This probe is available without authentication, so it should not contain any data.
}
```

So our ideal error message could be "Problems with: Prometheus, Grafana.", not more. And error message like "not ready" or "error" is better than more details.